sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -391,19 +393,38 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
 
         .. versionadded:: 0.22
 
+    as_frame : bool, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string, or categorical). The target is
+        a pandas DataFrame or Series depending on the number of
+        `target_columns`.
+
+        .. note::
+            Since the dataset is sparse, loading it as a pandas DataFrame
+            requires pandas 0.25+.
+
+        .. versionadded:: 0.24
+
     Returns
     -------
     bunch : :class:`~sklearn.utils.Bunch`
         Dictionary-like object, with the following attributes.
 
-        data: sparse matrix of shape (n_samples, n_features)
-            The data matrix to learn.
-        target: ndarray of shape (n_samples,)
-            The target labels.
+        data: {sparse matrix, dataframe} of shape (n_samples, n_features)
+            The data matrix to learn. If ``as_frame`` is `True`, ``data`` is
```

Comment:
I know it's not a part of this PR, but I find `The data matrix to learn` a bit odd. Maybe `The input data matrix`?